### PR TITLE
Issue 6 resources

### PR DESCRIPTION
**Pull request by Roccubalbetta**

### CODE REVIEW

**Review by jackmounts — CHANGES_REQUESTED** [review]
.classhpath e .settings nel gitignore pls cosi non si rompe sempre

**Comment by eutampieri** [comment]
Nei prossimi giorni faccio un post sul forum perché non dovrebbe rompersi in questo modo

**Review by jackmounts — APPROVED** [review]

File: src/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+		
+		// world objects
+		allImages.put("coin", tileSheet.cutImage(0, 144, 16,16));
+		allImages.put("skull", tileSheet.cutImage(16, 144, 16,16));
+		allImages.put("fire", tileSheet.cutImage(0, 160, 8, 21));
+		allImages.put("potion", tileSheet.cutImage(0,181,9,11));
+		allImages.put("gun", tileSheet.cutImage(0, 193, 21, 17));
+	}
+	
+	public void loadPlayerAnimations() {
```
Comment (eutampieri) [inline src/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java:93]: I would add an `Animatable` interface and implement it for the necessary objects here

File: src/main/java/eu/eutampieri/catacombs/ui/utils/FontUtils.java [hunk 2]
```diff
@@ -6,8 +6,7 @@
 public class FontUtils {
 	
 	public static int getTextWidth(Font font, String text) {
-		return (int) font.getStringBounds(text, new FontRenderContext(font.getTransform(),
-				true, true)).getWidth();
+		return (int) font.getStringBounds(text, new FontRenderContext(font.getTransform(), true, true)).getWidth();
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/ui/utils/FontUtils.java:9]: Maybe add a few variables for clarity to break this one-liner

**Review by eutampieri — CHANGES_REQUESTED** [review]

File: src/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java (hunk 1 above)
Comment (Roccubalbetta) [reply src/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java]: fixed

File: src/main/java/eu/eutampieri/catacombs/ui/utils/FontUtils.java (hunk 2 above)
Comment (Roccubalbetta) [reply src/main/java/eu/eutampieri/catacombs/ui/utils/FontUtils.java:9]: fixed but in issue_6_utils_loaders

File: src/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java (hunk 1 above)
Comment (eutampieri) [reply src/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java]: Where?

File: src/main/java/eu/eutampieri/catacombs/ui/utils/FontUtils.java (hunk 2 above)
Comment (eutampieri) [reply src/main/java/eu/eutampieri/catacombs/ui/utils/FontUtils.java]: Well, we’ll have to wait for the pr to be merged

File: src/main/java/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+		loadBossAnimations("Boss_Idle_right",  6, false, true);
+		loadBossAnimations("Boss_Idle_left",  6, true, true);
+		loadBossAnimations("Boss_Walk_right",  6, false, false);
+		loadBossAnimations("Boss_Walk_right",  6, true, false);
+
+		loadGunAnimations("Projectile_1", GUN_SHEET, 3, 1226, 64, 26);
+		loadGunAnimations("Projectile_2", GUN_SHEET, 5, 1418, 46, 118);
+		loadGunAnimations("Projectile_3", GUN_SHEET, 3, 1798, 37, 1187);
+		loadGunAnimations("Projectile_4", GUN_SHEET, 5, 2636, 64, 0);
+
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java:73]: I think this is really bad!

Comment (eutampieri) [reply src/main/java/eu/eutampieri/catacombs/ui/gamefx/AssetManager.java:73]: You should map a list of model object into a list of assets using streams

**Review by eutampieri — CHANGES_REQUESTED** [review]

**Comment by eutampieri** [comment]
I’m going to merge this but I’d like to see another one where you use a TOML file to load assets

**Pull request merged**